### PR TITLE
Do not change lastTree when moving a volatile RegStore node

### PR DESCRIPTION
**Pull request by a7ehuo**
When a volatile RegStore node is moved down to the end of the block,
the order of the trees cannot be changed because that is the order
containsNode used to check if canMoveIfVolatile should be true or not

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@vijaysun-omr @0xdaryl May I ask you to review this change? Thank you!

File: compiler/optimizer/DeadTreesElimination.cpp [hunk 1]
```diff
@@ -1041,7 +1041,35 @@ int32_t TR::DeadTreesElimination::process(TR::TreeTop *startTree, TR::TreeTop *e
...
+               // TT3 <--- lastTree
+               //
+               // === After adjusting lastTree & prevLastTree
+               // TT1 <--- prevLastTree
+               // TTBeingMoved (Volatile regStore) <--- Incorrect!!!
+               // TT2 (Volatile regStore) <--- lastTree
+               // TT3
+
+               bool skipMoveLastTreeIfVolatile = false;
+               if (iter.currentTree()->getNode()->getOpCode().isStoreReg() &&
```
Comment (0xdaryl) [inline compiler/optimizer/DeadTreesElimination.cpp:1065]: This is likely due to my unfamiliarity with this code, but is this phenomenon only an issue for regStores and not field stores?

Comment (a7ehuo) [reply compiler/optimizer/DeadTreesElimination.cpp:1065]: That's a good question. This change will only be run when the node (`isStoreReg` && RC > 1). If the node is not StoreReg, [the node is removed](https://github.com/eclipse/omr/blob/3cad1dcdcdc6867490787496e2b0a3efa439e7e1/compiler/optimizer/DeadTreesElimination.cpp#L985-L991). I think the check on `isStoreReg` is redundant. `node` here is `iter.currentTree()->getNode()`. I think I can also simplify the code to just check `node->getFirstChild()->mightHaveVolatileSymbolReference()` directly here. 

```
         if (!node->getOpCode().isStoreReg() || (node->getFirstChild()->getReferenceCount() == 1))
            {
            // Actually going to remove the treetop now
            //
            if (performTransformation(comp(), "%sRemove tree : [" POINTER_PRINTF_FORMAT "] ([" POINTER_PRINTF_FORMAT "] = %s)\n", optDetailString(), node, node->getFirstChild(), node->getFirstChild()->getOpCode().getName()))
               {
               ...
               }
            }
         else
            {
            if (performTransformation(comp(), "%sMove tree : [" POINTER_PRINTF_FORMAT "]([" POINTER_PRINTF_FORMAT "] = %s) to end of block\n", optDetailString(), node, node->getFirstChild(), node->getFirstChild()->getOpCode().getName()))
               {
               ...
               // This change 
               ... 
               }
            }
```

Comment (a7ehuo) [reply compiler/optimizer/DeadTreesElimination.cpp]: @0xdaryl @vijaysun-omr updated in d08229b. Ready for another review

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by vijaysun-omr — APPROVED** [review]

**Pull request merged**